Syntax.jj

```diff
@@ -1,41 +1,79 @@
 options{
-LOOKAHEAD = 1;
+LOOKAHEAD = 2;
 STATIC = false ;
 }
-PARSER_BEGIN(Adder)
-    class Adder{
+PARSER_BEGIN(Jason)
+    class Jason{
     public static void main(String[] args)
         throws ParseException, TokenMgrError{
-        Adder parser = new Adder(System.in);
-        parser.Start();
+        Jason parser = new Jason(System.in);
+        parser.Program();
 
         }
 
     }
-PARSER_END(Adder)
+PARSER_END(Jason)
 
-    SKIP: {" "}
-    SKIP: {"\n" | "\r" | "\r\n"}
-    TOKEN: {<PLUS: "+">}
+    SKIP: {" " | "\n" | "\r" | "\r\n" | "\t"}
     TOKEN: {<NUM: (["0"-"9"])+>}
     TOKEN: {<PROGRAM: "program">}
     TOKEN: {<BEGIN: "begin">}
     TOKEN: {<END: "end">}
-    TOKEN: {<EOL: ";">} 
-    TOKEN: {<IDENTIFIER: (["a"-"z", "A"-"Z"])+>}
-    TOKEN: {<ARRAY: "array">}
-    TOKEN: {<LBR: "[">}
-    TOKEN: {<RBR: "]">}
+    TOKEN: {<READ: "read">}
+    TOKEN: {<WRITE: "write">}
+    TOKEN: {<IF: "if">}
+    TOKEN: {<SET: "set">}
+    TOKEN: {<ENDIF: "endif">}
+    TOKEN: {<ELSE: "else">}
+    TOKEN: {<DO: "do">}
+    TOKEN: {<WHILE: "while">}
+    TOKEN: {<ENDWHILE: "endwhile">}
+    TOKEN: {<UNTIL: "until">}
+    TOKEN: {<ENDUNTIL: "enduntil">}
+    TOKEN: {<CALL: "call">}
     TOKEN: {<RECORD: "record">}
     TOKEN: {<TYPES: "types">}
     TOKEN: {<VARIABLES: "variables">}
     TOKEN: {<REAL: "real">}
     TOKEN: {<INT: "integer">}
-    TOKEN: {<STRING: "\"" (<IDENTIFIER> | <NUMBERS> | " ")+ "\"">}
+    TOKEN: {<PROC: "procedure">}
+    TOKEN: {<PARAM: "parameters">}
+    TOKEN: {<VAR: "var">}
+    TOKEN: {<FUNCTION: "function">}
+    TOKEN: {<RETURNS: "returns">}
+    TOKEN: {<RETURN: "return">}
+    TOKEN: {<THEN: "then">}
+    TOKEN: {<ARRAY: "array">}
+    TOKEN: {<EOL: ";">}
+    TOKEN: {<IDENTIFIER: ("_")* (["a"-"z", "A"-"Z", "_"])+ (<NUM>)*>}
```

Comment:
I guess this isn't the proposed behavior. 
`_8123123` should be an okay variable name, but in this way it is not accepted.
What we want is something starting with a letter or a `"_"` and then anything valid after.
Something like this:   
`{<IDENTIFIER: (["a"-"z", "A"-"Z", "_"]) ["a"-"z", "A"-"Z", "_",<NUM>] + >}`

Ps: Not sure if writting `<NUM>` inside the `[]` is valid.

Reply:
Parabéns aos envolvidos